### DISCUSSION TITLE
About Github IP Restriction & Github Apps

### DISCUSSION THREAD

**Discussion by nysdin** [community/community#69973] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am thinking of enabling IP restrictions on Github.
If I'm integrating with Github Apps, is it correct to understand that in order to automatically add the IP address of Github Apps to the IP Allow List, the IP address needs to be added to IP Allow List settings on Github Apps?

When using third-party Github Apps, if the Github Apps creator hasn't set up the IP Allow List or hasn't made the IP public, is it correct to understand that I won't be able to add to the IP allow list and hence, won't be able to use such Github Apps?

https://docs.github.com/en/enterprise-cloud@latest/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/managing-allowed-ip-addresses-for-your-organization#allowing-access-by-github-apps

**Comment by oussamaef06** [comment]
GitHub allows organizations to manage IP restrictions for added security. Here's how IP restrictions work in relation to GitHub Apps and third-party GitHub Apps:

GitHub Apps and IP Allow List:

GitHub Apps are integrated with GitHub repositories and organizations and are authorized to perform actions on your behalf. The IP addresses used by GitHub Apps are generally managed by GitHub.
IP Allow List for GitHub Apps:

GitHub Apps, like other GitHub services, have a range of IP addresses that they use for their operations. If you want to allow access only to GitHub Apps, you can add the specific IP addresses used by GitHub Apps to your IP Allow List. These IP addresses should be publicly available or provided by GitHub.
Third-Party GitHub Apps:

When you integrate third-party GitHub Apps, the situation can be different. The creator of the third-party GitHub App is responsible for its infrastructure and how it interacts with GitHub. GitHub does not manage or control the IP addresses used by these third-party Apps.
IP Restrictions and Third-Party Apps:

If the creator of a third-party GitHub App has not set up an IP Allow List or hasn't made their IP addresses publicly available, you won't be able to add those IP addresses to your organization's IP Allow List. You can only add IP addresses that you have control over and can verify.
GitHub Apps Permissions:

When installing a third-party GitHub App, you should review the permissions and access it requests. Ensure that you trust the app and its creator before granting it access to your organization's repositories.
Security Considerations:

When dealing with third-party GitHub Apps, it's essential to consider the security implications and trustworthiness of the app and its creator, as GitHub cannot vouch for the behavior of third-party Apps in the same way it can for its own services.
Remember that GitHub's features and policies may change over time, so it's a good idea to review GitHub's official documentation and security settings for the most up-to-date information. If you have specific concerns about third-party GitHub Apps, you can also reach out to the app's creator for clarification on their IP address and security practices.
